components/skus/browser/rs/wasm/src/lib.rs

```diff
@@ -137,21 +133,31 @@ impl JSSDK {
         let sdk = self.sdk.clone();
 
         let future = async move {
-            let window: ChromeWindow = web_sys::window()
-                .ok_or("couldn't get window")?
-                .dyn_into()
-                .unwrap();
+            let window: ChromeWindow =
+                web_sys::window().ok_or("couldn't get window")?.dyn_into().unwrap();
             if let Some(chrome) = window.chrome() {
                 if let Some(skus) = chrome.braveSkus() {
                     return JsFuture::from(skus.refresh_order(&order_id)).await;
                 }
             }
 
-            let order = sdk
-                .refresh_order(&order_id)
-                .await
-                .map_err(|e| e.to_string())?;
-            Ok(JsValue::from_serde(&order).unwrap())
+            let order = sdk.refresh_order(&order_id).await.map_err(|e| JsError::from(e))?;
```

Comment:
When you use a class method like this, you can just `map_err(JsError::from)`

Reply:
https://github.com/brave/brave-core/pull/13820/commits/448432a1248d454e47559d00032f42420812f015